subsys/fs/fat_fs.c

```diff
@@ -119,6 +119,7 @@ static int fatfs_close(struct fs_file_t *zfp)
 	return translate_error(res);
 }
 
+#if !defined(CONFIG_FS_FATFS_READ_ONLY)
```

Comment:
This looks related to the other proposal; I think it'd be totally reasonable to implement this as:
```
static int fatfs_unlink(struct fs_mount_t *mountp, const char *path)
￼{
    FRESULT res = -ENOTSUP;
#if !defined(CONFIG_FS_FATFS_READ_ONLY)
    // main body
    res = translate_error(res);
#endif
    return res;
}
```

which eliminates the need to make any changes at all to the `fs_file_system_t` objects.

Reply:
> 
> 
> This looks related to the other proposal; I think it'd be totally reasonable to implement this as:
> 
> ```
> static int fatfs_unlink(struct fs_mount_t *mountp, const char *path)
> ￼{
>     FRESULT res = -ENOTSUP;
> #if !defined(CONFIG_FS_FATFS_READ_ONLY)
>     // main body
>     res = translate_error(res);
> #endif
>     return res;
> }
> ```
> 
> which eliminates the need to make any changes at all to the `fs_file_system_t` objects.

I can do that, no problem for me.

The #29396  has not been created because of this. Basically I wanted to get rid of the NULL check within fs_ functions, and I have offered something other in exchange.
You may not agree with me but I think the NULL checks are not needed there as I believe that developers exactly know which functions are offered by the system driver or at least will gain the knowledge quickly.
Basically: when writing embedded software you get -ENOTSUP, specifically in the meaning "not implemented". once  then you have to reorganize to solve the problem and you newer see it again.
If you green light me in #29396 to drop defaulting not implemented functions, while being allowed to remove/make optional the NULL checks: that will be exactly what I will do.